endpoints/s3/src/getobject.cpp

```diff
@@ -0,0 +1,339 @@
+            std::time_t last_write_time__time_t = std::chrono::system_clock::to_time_t(last_write_time__time_point);
+            std::string last_write_time__str = irods::s3::api::common_routines::convert_time_t_to_str(last_write_time__time_t, date_format);
+            persistent_data_ptr->response.insert(beast::http::field::last_modified, last_write_time__str);
+
+            // seek to the start range
+            persistent_data_ptr->d.seekg(range_start);
+            size_t offset = range_start;
+
+            if (persistent_data_ptr->d.fail() || persistent_data_ptr->d.bad()) {
+                logging::error("{}: Fail/badbit set", __FUNCTION__);
+                persistent_data_ptr->response.result(beast::http::status::forbidden);
+                persistent_data_ptr->response.body().more = false;
+                logging::debug("{}: returned {}", __FUNCTION__, persistent_data_ptr->response.reason());
+                session_ptr->send(std::move(persistent_data_ptr->response)); 
+                return;
+            }
+            beast::error_code ec;
+            beast::http::write_header(session_ptr->stream().socket(), persistent_data_ptr->serializer, ec);
+            if (ec) {
+                persistent_data_ptr->response.result(beast::http::status::internal_server_error);
+                logging::debug("{}: returned {}", __FUNCTION__, persistent_data_ptr->response.reason());
+                session_ptr->send(std::move(persistent_data_ptr->response)); 
+                return;
+            }
+
+            // set the response to ok 
+            persistent_data_ptr->response.result(beast::http::status::ok);
+
+            read_from_irods_send_to_client(
+                session_ptr,
+                persistent_data_ptr,
+                range_start,
+                range_end,
+                offset,
+                write_buffer_size,
+                __FUNCTION__);
+        }
+        else {
+            return irods::s3::api::common_routines::send_error_response(
+                    session_ptr,
+                    boost::beast::http::status::not_found,
+                    "NoSuchKey",
+                    "Object does not exist",
+                    url.path(),
+                    __FUNCTION__);
+        }
+    }
+    catch (irods::exception& e) {
+        logging::error("{}: Exception {}", __FUNCTION__, e.what());
+
+        switch (e.code()) {
+            case USER_ACCESS_DENIED:
+            case CAT_NO_ACCESS_PERMISSION:
+                response.result(beast::http::status::forbidden);
+                break;
+            default:
+                response.result(beast::http::status::internal_server_error);
+                break;
+        }
+
+        logging::debug("{}: returned {}", __FUNCTION__, response.reason());
+        session_ptr->send(std::move(response)); 
+        return;
```

Comment:
This `return` statements seem unnecessary. Consider removing it.

_There are others too._